### PR TITLE
Fixups on top of PR #1204

### LINKED ISSUE #1200: [ttGlyphPen] decompose components if transform overflows F2Dot14
https://github.com/googlei18n/ufo2ft/issues/217

The UFO GLIF spec allows any numbers for xScale, xyScale, yxScale, yScale, xOffset, yOffset, however the OpenType glyf spec uses F2Dot14 numbers, which are encoded as a signed 16-bit integer and therefore can only contain values from -32768 (-0x8000, or -2.0) to +32767 included (0x7FFF, or +1.99993896484375...).

We can't let the `struct.error` propagate.

I think we have to handle the case of +2.0 specially, and treat it as if it were 1.99993896484375. By doing that we can support truetype component transforms in the range -2.0 to +2.0 (inclusive), for the sake of simplicity.

Then, we also need to have the ttGlyphPen decompose the components if their transform values are less than -2.0 or they are greater than +2.0 (not greater and equal), as these can't fit in the TrueType glyf table.

### PR DESCRIPTION
**Pull request by anthrotype**
this is a follow up from @moyogo's PR #1204 (after a private conversation).

Fixes #1200

### CODE REVIEW

File: Lib/fontTools/pens/ttGlyphPen.py [hunk 1]
```diff
@@ -91,22 +88,18 @@ def glyph(self, componentFlags=0x4):
         assert self._isClosed(), "Didn't close last contour."
 
         for i, (glyphName, transformation) in enumerate(self.components):
```
Comment (anthrotype) [inline Lib/fontTools/pens/ttGlyphPen.py]: ah! we sholdn't modify a list that we are iterating on.. i'll fix it

**Comment by behdad** [comment]
I'm not sure doing this in TTGlyphPen doesn't have consequences. In particular in the varfont workflow. IMO this should be done in ufo2ft, and done to all masters the same way.

**Comment by anthrotype** [comment]
yes, that will need to be done anyway when building interpolatable TTFs. But for the general case when a TTGlyphPen is sent arbitrary components' transforms (without knowing where they come from and what is the purpose), then it will do the right thing and decompose when they exceed the F2Dot14 bounds. It's always better than raise an error.

**Comment by anthrotype** [comment]
unless... we are happy with raising a struct.error here and leave it to the client (in our case, ufo2ft) to deal with it the way they deem appropriate (e.g. clamp to almost 2, decompose if out of range).

**Comment by behdad** [comment]
Make it an option then, detail to True is fine, but disable in interpolatable output.

**Comment by anthrotype** [comment]
> disable in interpolatable output

so we don't have to check twice?

**Comment by anthrotype** [comment]
i'm confused: when you say "default to True" do you mean raise the error by default, or "clamp and decompose" by default? I'd say the latter.

**Comment by behdad** [comment]
Default whatever you want. Just disable decompose in interpolatablr output.

**Comment by anthrotype** [comment]
ok

**Comment by anthrotype** [comment]
ok, I guarded both the decomposition and the clamping-to-2 with an option (called `handleOverflowingTransforms`), defaults to True. Going to merge this once the bots are happy

**Pull request merged**